blockchain/v2/scheduler.go

```diff
@@ -383,3 +469,107 @@ func (sc *schedule) numBlockInState(targetState blockState) uint32 {
 	}
 	return num
 }
+
+// Handlers
+
+// XXX: This handler will probably get the block first and then pass it on to the processor
+func (sc *scheduler) handleBlockResponse(event tmpBlockResponse) (Event, error) {
+	err := sc.touchPeer(event.peerID, event.time)
+	if err != nil {
+		return scPeerError{peerID: event.peerID, reason: err}, nil
+	}
+
+	err = sc.markReceived(event.peerID, event.block.Height, event.size, event.time)
+	if err != nil {
+		return scPeerError{peerID: event.peerID, reason: err}, nil
+	}
+
+	return scBlockReceived{peerID: event.peerID, block: event.block}, nil
+}
+
+func (sc *scheduler) handleBlockProcessed(event pcBlockProcessed) (Event, error) {
+	err := sc.markProcessed(event.height)
+	if err != nil {
+		return scSchedulerFail{reason: err}, nil // this should be fatal
```

Comment:
It is possible that a peer error or timeout is handled after the processor has processed the block but before the scheduler received `pcBlockProcessed` (prio normal) from the processor. When `pcBlockProcessed` event is received the block might be marked requested but not yet received. 
Not sure what the best solution is here, maybe just ignore the error from `markProcessed()` since we know that the block has been verified and applied to the state, in the store etc.